### PR TITLE
Fix loading metadata cache with path separator prefix

### PR DESCRIPTION
**Pull request by nopcoder**
Fix https://github.com/treeverse/lakeFS/issues/5921

### CODE REVIEW

**Review by Jonathan-Rosenberg — APPROVED** [review]
How was this tested?

**Comment by nopcoder** [comment]
> How was this tested?

Manually running lakeFS with local folder holding files - the cache was filled with unused keys.

**Review by arielshaqed — COMMENTED** [review]
This is unspeakably horrible and beautiful at the same time 🤯 

**Great** catch!

File: pkg/pyramid/tier_fs.go [hunk 1]
```diff
@@ -85,7 +81,13 @@ func (tfs *TierFS) log(ctx context.Context) logging.Logger {
 //  2. Remove workspace directories and all its content if it
 //     exists under the namespace dir.
 func (tfs *TierFS) handleExistingFiles() error {
-	if err := filepath.Walk(tfs.fsLocalBaseDir, func(p string, info os.FileInfo, err error) error {
+	// setup cache dir location with separator as suffix. store are relative without prefix.
+	dir := tfs.fsLocalBaseDir
+	if !strings.HasSuffix(dir, string(filepath.Separator)) {
+		dir += string(filepath.Separator)
```
Comment (arielshaqed) [inline pkg/pyramid/tier_fs.go:87]: `filepath.Separator` might not be `/`, but we always do want that slash!

```suggestion
	if !strings.HasSuffix(dir, "/") {
		dir += "/"

**Review by arielshaqed — COMMENTED** [review]
Thinking about it, code is broken on Windows -- and probably not in a nice way: it transforms prefix "`s3://bucket/storage/namespace/`" to "`s3://bucket/storage/namespace/\`", which will probably break things.

Not sure how many Windows installations we have, but this example makes me think that using filepath.Separator is incorrect.

**Pull request merged**